src/pudl/transform/vcerare.py

```diff
@@ -0,0 +1,330 @@
+"""Transformations of the Vibrant Clean Energy renewable generation profiles.
+
+Wind and solar profiles are extracted separately, but concatenated into a single table
+in this module, as they have exactly the same structure.
+"""
+
+import pandas as pd
+from dagster import AssetCheckResult, asset, asset_check
+
+import pudl
+from pudl.helpers import cleanstrings_snake, simplify_columns, zero_pad_numeric_string
+from pudl.metadata.dfs import POLITICAL_SUBDIVISIONS
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def _prep_lat_long_fips_df(raw_vcegen__lat_lon_fips: pd.DataFrame) -> pd.DataFrame:
+    """Prep the lat_long_fips table to merge into the capacity factor tables.
+
+    Prep entails making sure the formatting and column names match those in the
+    capacity factor tables, adding 0s to the beginning of FIPS codes with 4 values,
+    and making separate county/subregion and state columns.
+
+    The county portion of the county_state column does not map directly to FIPS ID.
+    Some of the county names are actually subregions like cities or lakes. For this
+    reason we've named the column county_or_lake_name and it should be considered
+    part of the primary key. There are several instances of multiple subregions that
+    map to a single county_id_fips value.
+    """
+    logger.info(
+        "Preping Lat-Long-FIPS table for merging with the capacity factor tables"
+    )
+    ps_usa_df = POLITICAL_SUBDIVISIONS[POLITICAL_SUBDIVISIONS["country_code"] == "USA"]
+    state_names = cleanstrings_snake(
+        ps_usa_df, ["subdivision_name"]
+    ).subdivision_name.tolist()
+    state_pattern = "|".join(state_names)
+    # Transform lat long fips table by making the county_state_names lowercase
+    # to match the values in the capacity factor tables. Fix FIPS codes with
+    # no leading zeros, and add a county_or_lake_name and state field.
+    lat_long_fips = (
+        raw_vcegen__lat_lon_fips.pipe(simplify_columns)
+        .assign(
+            county_state_names=lambda x: x.county_state_names.str.lower().replace(
+                {r"\.": "", "-": "_"}, regex=True
+            )
+        )
+        .assign(county_id_fips=lambda x: zero_pad_numeric_string(x.fips, 5))
+        .assign(state_id_fips=lambda x: x.county_id_fips.str.extract(r"(\d{2})"))
+        .assign(
+            county_or_lake_name=lambda x: x.county_state_names.str.extract(
+                rf"(.+)_({state_pattern})$"
```

Comment:
Is there any reason not to more narrowly capture [a-z]+ here, instead of .+?

Reply:
no reason! think I didn't want to run the risk of there being a hyphen or something in there. These county names are relatively informal, so whatever comes before the state pattern is what we are taking. If we wanted something more official we could just merge the County Name from our dp1 census table.